modules/shared-vpc-policy-and-hosted-zone/main.tf

```diff
@@ -0,0 +1,111 @@
+locals {
+  name_prefix = var.name_prefix != null ? var.name_prefix : var.cluster_name
+}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_region" "current" {}
+
+resource "aws_iam_role" "shared_vpc_role" {
+  name = "${local.name_prefix}-shared-vpc-role"
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/ResourceGroupsandTagEditorFullAccess"
+  ]
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Sid    = ""
+        Principal = {
+          AWS = [
+            var.ingress_operator_role_arn,
+            var.installer_role_arn
+          ]
+        }
+      }
+    ]
+  })
+  description = "Role that will be assumed from the Target AWS account where the cluster resides"
+  lifecycle {
+    ignore_changes = [managed_policy_arns]
+  }
+}
+
+resource "aws_iam_policy" "shared_vpc_policy" {
+  name = "${local.name_prefix}-shared-vpc-policy"
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "route53:ChangeResourceRecordSets",
+          "route53:ListHostedZones",
+          "route53:ListHostedZonesByName",
+          "route53:ListResourceRecordSets",
+          "route53:ChangeTagsForResource",
+          "route53:GetAccountLimit",
+          "route53:GetChange",
+          "route53:GetHostedZone",
+          "route53:ListTagsForResource",
+          "route53:UpdateHostedZoneComment",
+          "tag:GetResources",
+          "tag:UntagResources"
+        ]
+        Resource = "*"
```

Comment:
Can you be more specific here?
I'm not sure but please check

Reply:
I change it a bit
Anyway, we will have to review this to understand if this is the minimal permissions can be added
Opened ticket [OCM-5663](https://issues.redhat.com/browse/OCM-5663)